src/promql/src/functions/quantile_aggr.rs

```diff
@@ -0,0 +1,298 @@
+    }
+
+    fn evaluate(&mut self) -> DfResult<ScalarValue> {
+        let values: Vec<_> = self.values.iter().map(|v| v.unwrap_or(0.0)).collect();
+
+        let result = quantile_impl(&values, self.q);
+
+        ScalarValue::new_primitive::<Float64Type>(result, &DataType::Float64)
+    }
+
+    fn size(&self) -> usize {
+        std::mem::size_of::<Self>() + self.values.capacity() * std::mem::size_of::<Option<f64>>()
+    }
+
+    fn state(&mut self) -> DfResult<Vec<ScalarValue>> {
+        let values_array = Arc::new(ListArray::from_iter_primitive::<Float64Type, _, _>(vec![
+            Some(self.values.clone()),
+        ]));
+
+        let state_struct = StructArray::new(
+            vec![Field::new(
+                VALUES_FIELD_NAME,
+                DataType::List(Arc::new(Field::new(
+                    DEFAULT_LIST_FIELD_NAME,
+                    DataType::Float64,
+                    true,
+                ))),
+                false,
+            )]
+            .into(),
+            vec![values_array],
+            None,
+        );
+
+        Ok(vec![ScalarValue::Struct(Arc::new(state_struct))])
+    }
+
+    fn merge_batch(&mut self, states: &[ArrayRef]) -> DfResult<()> {
+        if states.is_empty() {
+            return Ok(());
+        }
+
+        for state in states {
+            let state = as_struct_array(state)?;
+
+            for list in as_list_array(state.column(0))?.iter().flatten() {
+                let f64_array = as_primitive_array::<Float64Type>(&list)?.clone();
+                self.values.extend(&f64_array);
+            }
+        }
+
+        Ok(())
+    }
+}
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use datafusion::arrow::array::{ArrayRef, Float64Array};
+    use datafusion_common::ScalarValue;
+
+    use super::*;
+
+    fn create_f64_array(values: Vec<Option<f64>>) -> ArrayRef {
+        Arc::new(Float64Array::from(values)) as ArrayRef
+    }
+
+    #[test]
+    fn test_quantile_accumulator_empty() {
+        let mut accumulator = QuantileAccumulator::new(0.5);
+
+        let result = accumulator.evaluate().unwrap();
+
+        match result {
+            ScalarValue::Float64(_) => (),
+            _ => panic!("Expected Float64 scalar value"),
+        }
+    }
+
+    #[test]
+    fn test_quantile_accumulator_single_value() {
+        let mut accumulator = QuantileAccumulator::new(0.5);
+        let input = create_f64_array(vec![Some(10.0)]);
+
+        accumulator.update_batch(&[input]).unwrap();
+        let result = accumulator.evaluate().unwrap();
+
+        assert_eq!(result, ScalarValue::Float64(Some(10.0)));
+    }
+
+    #[test]
+    fn test_quantile_accumulator_multiple_values() {
+        let mut accumulator = QuantileAccumulator::new(0.5);
+        let input = create_f64_array(vec![Some(1.0), Some(2.0), Some(3.0), Some(4.0), Some(5.0)]);
+
+        accumulator.update_batch(&[input]).unwrap();
+        let result = accumulator.evaluate().unwrap();
+
+        // 中位数应该是 3.0
```

Comment:
👀 maybe you can try prompting in English

Reply:
hmmm, forgot to delete it.